torch_xla/csrc/tensor.cpp

```diff
@@ -1405,6 +1405,33 @@ void XLATensor::masked_fill_(XLATensor& input, const XLATensor& mask,
                                                      expanded_mask, value));
 }
 
+XLATensor XLATensor::cross(const XLATensor& input, const XLATensor& other,
+                           xla::int64 dim) {
+  if (dim < 0) {
+    auto input_shape_ref = input.shape();
+    auto dim_3_it = std::find((*input_shape_ref).dimensions().begin(),
+                              (*input_shape_ref).dimensions().end(), 3);
+    XLA_CHECK(dim_3_it != (*input_shape_ref).dimensions().end())
+        << "No dimension of size 3 in input: " << (*input_shape_ref).ToString();
+    dim = dim_3_it - (*input_shape_ref).dimensions().begin();
+  }
+  XLA_CHECK_EQ(input.size(dim), 3)
+      << "Invalid cross argument: dimension " << dim << " does not have size 3";
+  XLA_CHECK_LT(dim, input.shape().get().rank())
+      << "Invalid cross argument: dimension " << dim << " out of range";
+  std::vector<std::pair<xla::int64, xla::int64>> cross_product_dim_pairs = {
+      {1, 2}, {2, 0}, {0, 1}};
+  std::vector<XLATensor> xyz;
```

Comment:
I find it a bit more clear the one at tensorflow/compiler/tf2xla/kernels/cross_op.cc:57
Which also create less operations by caching the slices.

Reply:
I can make it a bit more explicit and cache the slice as well without going all the way to xla operations, I don't find that readable.

Reply:
Done.